crates/uv-git/src/git.rs

```diff
@@ -230,15 +230,23 @@ impl GitRemote {
         locked_rev: Option<GitOid>,
         client: &ClientWithMiddleware,
         disable_ssl: bool,
+        offline: bool,
```

Comment:
Could we just pass the connectivity into here instead of a bool?

Reply:
I guess we pass bools for `disable_ssl`, but I presume that's because the type itself is more complex. I don't feel strongly, but less bools is nice.

Reply:
@zanieb will defer to you! only passed in as a bool because it meant not having to add an additional import to `uv-git` to pull in the type (`Connectivity` is defined in `uv-client`)